src/components/dataset/query/sidebar/QueryViewSidebarItem.jsx

```diff
@@ -39,37 +62,56 @@ export class QueryViewSidebarItem extends React.PureComponent {
       tableName,
       token,
     } = this.props;
-    switch (column.datatype) {
-      case 'string':
-        return (
-          <CategoryWrapper
-            column={column}
-            dataset={dataset}
-            filterData={filterData}
-            filterStatement={filterStatement}
-            joinStatement={joinStatement}
-            handleChange={this.handleChange}
-            handleFilters={handleFilters}
-            tableName={tableName}
-            token={token}
-          />
-        );
-      case 'float':
-      case 'integer':
-        return (
-          <RangeFilter
-            column={column}
-            dataset={dataset}
-            filterData={filterData}
-            handleChange={this.handleChange}
-            handleFilters={handleFilters}
-            tableName={tableName}
-            token={token}
-          />
-        );
-      default:
-        return <div />;
-    }
+    const { disableButton, filterMap } = this.state;
+    const item =
+      column.datatype === 'string' ? (
```

Comment:
If I understand correctly, if the column is returns strings, you get a filter where the rows are strings, otherwise if it's numeric you get a filter where the rows are selectable ranges?

Reply:
Yes! The datatype of the column determines the type of input component is used to filter the data. The user can select specific string values using checkboxes or entering them into a text box, whereas numeric values are filtered with a range slider.